lib/semantics/check-coarray.cc

```diff
@@ -46,6 +46,21 @@ void CoarrayChecker::Leave(const parser::FormTeamStmt &x) {
   CheckTeamType(context_, std::get<parser::TeamVariable>(x.t));
 }
 
+void CoarrayChecker::Enter(const parser::CriticalConstruct &x) {
+  auto &criticalStmt{
+      std::get<parser::Statement<parser::CriticalStmt>>(x.t)};
+
+  const parser::Block &block{std::get<parser::Block>(x.t)};
+  CriticalBodyEnforce criticalBodyEnforce{context_, criticalStmt.source};
+  parser::Walk(block, criticalBodyEnforce);
+
+  // C1119: A branch within a CRITICAL construct shall not have a branch target
```

Comment:
We usually just name the constraint (C1119) rather than including its full text.

Reply:
I will change this in the next set of changes.